src/strands/ir_builders.js

```diff
@@ -463,40 +464,72 @@ export function swizzleTrap(id, dimension, strandsContext) {
-          && target.dimension >= chars.length;
-        if (valid) {
-          const originalNode = DAG.getNodeDataFromID(strandsContext.dag, target.id);
-          const changed = Object.fromEntries(
-            chars.map((c, i) => [set.indexOf(c), i])
-          )
-          if (typeof value === 'number') {
-            value = new Array(chars.length).fill(value);
-          } 
-          const newValues = mapPrimitiveDepsToIDs(
-            strandsContext,
-            {baseType: originalNode.baseType, dimension: null}, 
-            value
-          ).mappedDependencies;
-          const newDeps = new Array(originalNode.dimension);
-          for (let i = 0; i < target.dimension; i++) {
-            console.log(changed[i])
-            if (changed[i] !== undefined) {
-              newDeps[i] = newValues[changed[i]];
-            }
+  set(target, property, value, receiver) {
+    for (const basis of swizzleSets) {
+      const chars = [...property];
+      const valid =
+        chars.every(c => basis.includes(c)) &&
+        new Set(chars).size === chars.length &&
+        target.dimension >= chars.length;
+
+      if (!valid) continue;
+
+      const dim = target.dimension;
+
+      const lanes = new Array(dim);
+      for (let i = 0; i < dim; i++) {
+        const { id, dimension } = swizzleNode(strandsContext, target, 'xyzw'[i]);
+        lanes[i] = createStrandsNode(id, dimension, strandsContext);
+      }
+
+      let scalars = [];
+      if (value instanceof StrandsNode) {
+        if (value.dimension === 1) {
+          scalars = Array(chars.length).fill(value);
+        } else if (value.dimension >= chars.length) {
+          for (let k = 0; k < chars.length; k++) {
+            const { id, dimension } = swizzleNode(strandsContext, value, 'xyzw'[k]);
+            scalars.push(createStrandsNode(id, dimension, strandsContext));
           }
-          console.log("AFTER: ", newDeps)
-          // primitiveConstructorNode(strandsContext, {baseType, dimension}, newDeps);
-          return true;
+        } else {
+          FES.userError('type error', `Swizzle assignment: RHS vector too short (need ${chars.length}, got ${value.dimension}).`);
+        }
+      } else if (Array.isArray(value)) {
+        const flat = value.flat(Infinity);
+        if (flat.length === 1) {
+          scalars = Array(chars.length).fill(flat[0]);
+        } else if (flat.length === chars.length) {
+          scalars = flat;
+        } else {
+          FES.userError('type error', `Swizzle assignment: RHS length ${flat.length} does not match ${chars.length}.`);
         }
+      } else if (typeof value === 'number') {
+        scalars = Array(chars.length).fill(value);
+      } else {
+        FES.userError('type error', `Unsupported RHS for swizzle assignment: ${value}`);
       }
-      return Reflect.set(...arguments);
+
+      for (let j = 0; j < chars.length; j++) {
+        const canonicalIndex = basis.indexOf(chars[j]);
```

Comment:
Possibly also worth a comment tying this back to the lane definition. e.g. map `x` and `r` to 0, `y` and `g` to 1, etc